panther_hardware_interfaces/test/test_panther_imu.cpp

```diff
@@ -0,0 +1,327 @@
+  EXPECT_NO_THROW({ imu_sensor_->CheckStatesSize(); });
+}
+
+TEST_F(TestPantherImuSensor, CheckInterfaces)
+{
+  hardware_interface::HardwareInfo info;
+  info.sensors.push_back({});
+
+  std::list<std::string> example_interfaces_names = {"state1", "state2", "state3", "state4",
+                                                     "state5", "state6", "state7", "state8",
+                                                     "state9", "state10"};
+  for (const auto & interface_name : example_interfaces_names) {
+    hardware_interface::InterfaceInfo interface_info;
+    interface_info.name = interface_name;
+    info.sensors.front().state_interfaces.push_back(interface_info);
+  }
+
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_THROW({ imu_sensor_->CheckInterfaces(); }, std::runtime_error);
+
+  info.sensors.front().state_interfaces.clear();
+  for (const auto & interface_name : panther_hardware_interfaces_test::kImuInterfaces) {
+    hardware_interface::InterfaceInfo interface_info;
+    interface_info.name = interface_name;
+    info.sensors.front().state_interfaces.push_back(interface_info);
+  }
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_NO_THROW({ imu_sensor_->CheckInterfaces(); });
+}
+
+TEST_F(TestPantherImuSensor, ReadObligatoryParams)
+{
+  hardware_interface::HardwareInfo info;
+  info.hardware_parameters = {{"param1", "value1"}, {"param2", "value2"}};
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_THROW({ imu_sensor_->ReadObligatoryParams(); }, std::exception);
+
+  info.hardware_parameters = panther_hardware_interfaces_test::kImuObligatoryParams;
+  info.hardware_parameters["callback_delta_epsilon_ms"] =
+    info.hardware_parameters["data_interval_ms"];
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_THROW({ imu_sensor_->ReadObligatoryParams(); }, std::exception);
+
+  info.hardware_parameters = panther_hardware_interfaces_test::kImuObligatoryParams;
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_NO_THROW({ imu_sensor_->ReadObligatoryParams(); });
+}
+
+TEST_F(TestPantherImuSensor, HandleFirstDataCallback)
+{
+  geometry_msgs::msg::Vector3 mag_compensated;
+  geometry_msgs::msg::Vector3 lin_acc;
+
+  mag_compensated.x = std::numeric_limits<float>::quiet_NaN();
+  mag_compensated.y = std::numeric_limits<float>::quiet_NaN();
+  mag_compensated.z = std::numeric_limits<float>::quiet_NaN();
+
+  // When imu is not calibrated  nans are skipped
+  EXPECT_NO_THROW({ imu_sensor_->HandleFirstDataCallback(mag_compensated, lin_acc, 0.0); });
+
+  mag_compensated.x = 1.0;
+  mag_compensated.y = 1.0;
+  mag_compensated.z = 1.0;
+
+  // IMU can't find gravity vector due to empyy acceleration vector
+  EXPECT_THROW(
+    { imu_sensor_->HandleFirstDataCallback(mag_compensated, lin_acc, 0.0); }, std::runtime_error);
+
+  // IMU should calibrate
+  lin_acc.z = 9.80665;
+  EXPECT_NO_THROW({ imu_sensor_->HandleFirstDataCallback(mag_compensated, lin_acc, 0.0); });
+}
+
+TEST(TestPantherImu, CheckInterfacesLoadedByResourceManager)
```

Comment:
This comment is in reference to the earlier suggestion regarding the relocation of PantherImuTestUtils to TestPantherImu. My preference is to transform these cases into a test suite and make use of the methods available in TestPantherImuSensor to achieve a more cohesive approach.